mono/metadata/threads.c

```diff
@@ -2266,7 +2266,7 @@ ves_icall_System_Threading_Thread_ClrState (MonoInternalThread* this_obj, guint3
 }
 
 void
-ves_icall_System_Threading_Thread_SetState (MonoInternalThread* this_obj, guint32 state)
+System_Threading_Thread_SetState_internal (MonoInternalThread* this_obj, guint32 state)
```

Comment:
Please rename to `mono_thread_set_state`

Reply:
That already exists. The difference is where is the event set for background. Move that into existing mono_thread_set_state? I considered that, but instead made the smaller more conservative change.

Reply:
Specifically, move this ```
	if (state & ThreadState_Background) {
		/* If the thread changes the background mode, the main thread has to
		 * be notified, since it has to rebuild the list of threads to
		 * wait for.
		 */
		mono_os_event_set (&background_change_event);
	}
``` into the lower level `set_thread_set_state`?